File: Som/Topology/GaussNeighbourhoodFunction.cs
using System;

namespace Som.Topology
{
    public class GaussNeighbourhoodFunction : INeighbourhoodFunction
    {
        public double GetDistanceFalloff(double distance, double radius)
        {
            double denominator = 2 * radius * radius;
            return Math.Exp(-(distance * distance / denominator));
        }
    }
}

File: JavaSom/Topology/GaussNeighbourhoodFunction.java
package Topology;

	public class GaussNeighbourhoodFunction implements INeighbourhoodFunction
	{
		public final double GetDistanceFalloff(double distance, double radius)
		{
			double denominator = 2 * radius * radius;
			return Math.exp(- distance / denominator);
		}
	}